src/SignalR/clients/ts/signalr-protocol-msgpack/src/MessagePackHubProtocol.ts

```diff
@@ -284,7 +284,8 @@ export class MessagePackHubProtocol implements IHubProtocol {
     }
 
     private _writeCompletion(completionMessage: CompletionMessage): ArrayBuffer {
-        const resultKind = completionMessage.error ? this._errorResult : completionMessage.result ? this._nonVoidResult : this._voidResult;
+        const resultKind = completionMessage.error ? this._errorResult :
```

Comment:
What if `completionMessage.error` is an empty string instead of `undefined` or `null`? Should that be treated as an error result still?

Reply:
Yeah, any non-null value should be considered an error. I don't think there is anyway today for an empty string to be set, but even if it was that should be considered an error.
In theory we should throw for both error and result being set, but since I want to backport this change, and we control the creation of completion messages it would basically only be an assert, I'm not going to add that.